### DISCUSSION TITLE
Configuring production branch release

### DISCUSSION THREAD

**Discussion by mrlubos** [semantic-release/semantic-release#3103] [Q&A]
Hey! I realise this might be a workflow question, but wanted to ask how to achieve the following.

We have a `main` branch which gets deployed to staging environment, and then we merge it into `production` branch to go live. `semantic-release` will create a release candidate on `main`, and actual release on `production`.

The problem I have is this. Let's say we're starting with version 1.0.0.
1. Push feature branch into `main`, create release `v1.0.1-rc.1` (good)
2. Push `main` branch into `production`, create release `v1.0.1` (good)
3. Push feature branch into `main`, create release `v1.0.1-rc.2` (uh oh!)

I understand why this happens. `semantic-release` bumps `package.json` on `main` to `v1.0.1-rc.1`, and when we deploy to production it bumps it to `v1.0.1`, but only on the `production` branch.

My question is, how to fix this so that when we release, `semantic-release` would bump the version in step 3 above to `v1.0.2-rc.1`?

**Comment by travi** [comment]
We do not recommend using semantic-release for a gitflow style workflow, but merging your production branch back into main should accomplish the versioning that you are targeting

**Reply by mrlubos** [reply]
Thank you @travi! Are there alternative tools you'd recommend?
